src/biogeochem/CNCIsoFluxMod.F90

```diff
@@ -1747,54 +1713,35 @@ subroutine CNCIsoGrossUnrepPftToColumn (num_soilc, filter_soilc, &
           )
 
        do j = 1, nlevdecomp
-          do pi = 1,maxsoil_patches
-             do fc = 1,num_soilc
-                c = filter_soilc(fc)
-                
-                if (pi <=  col%npatches(c)) then
-                   p = col%patchi(c) + pi - 1
-
-                   if (patch%active(p)) then
-
-                      do i = i_litr_min, i_litr_max
-                         gru_c_to_litr_c(c,j,i) = &
-                            gru_c_to_litr_c(c,j,i) + &
-                            ! leaf gross unrepresented landcover change mortality carbon fluxes
-                            gru_leafc_to_litter(p) * lf_f(ivt(p),i) * wtcol(p) * leaf_prof(p,j) + &
-                            ! fine root gross unrepresented landcover change mortality carbon fluxes
-                            gru_frootc_to_litter(p) * fr_f(ivt(p),i) * wtcol(p) * froot_prof(p,j)
-                      end do
-
-                      ! coarse root gross unrepresented landcover change mortality carbon fluxes
-                      gru_c_to_cwdc_c(c,j) = gru_c_to_cwdc_c(c,j) + &
-                           gru_livecrootc_to_litter(p) * wtcol(p) * croot_prof(p,j)
-                      gru_c_to_cwdc_c(c,j) = gru_c_to_cwdc_c(c,j) + &
-                           gru_deadcrootc_to_litter(p) * wtcol(p) * croot_prof(p,j) 
-
-                   end if
-                end if
+          do fp = 1,num_soilp
+             p = filter_soilp(fp)
+             c = patch%column(p)
 
+             do i = i_litr_min, i_litr_max
+                gru_c_to_litr_c(c,j,i) = &
+                   gru_c_to_litr_c(c,j,i) + &
+                   ! leaf gross unrepresented landcover change mortality carbon fluxes
+                   gru_leafc_to_litter(p) * lf_f(ivt(p),i) * wtcol(p) * leaf_prof(p,j) + &
+                   ! fine root gross unrepresented landcover change mortality carbon fluxes
+                   gru_frootc_to_litter(p) * fr_f(ivt(p),i) * wtcol(p) * froot_prof(p,j)
              end do
 
+             ! coarse root gross unrepresented landcover change mortality carbon fluxes
+             gru_c_to_cwdc_c(c,j) = gru_c_to_cwdc_c(c,j) + &
+                  gru_livecrootc_to_litter(p) * wtcol(p) * croot_prof(p,j)
+             gru_c_to_cwdc_c(c,j) = gru_c_to_cwdc_c(c,j) + &
+                  gru_deadcrootc_to_litter(p) * wtcol(p) * croot_prof(p,j) 
+
           end do
        end do
    
-       do pi = 1,maxsoil_patches
-          do fc = 1,num_soilc
-             c = filter_soilc(fc)
-
-             if (pi <=  col%npatches(c)) then
-                p = col%patchi(c) + pi - 1
-
-                if (patch%active(p)) then
```

Comment:
Ahhh, yes this is a nice improvement, just loop over the soil patches in the filter, and remove the if statements to make sure they are active and within range.